### PR TITLE
Fix MiMa configuration

### PR DESCRIPTION
**Pull request by marcospereira**
## Purpose

Follow up of #313. Fix MiMa configuration and other changes suggested by @ignasi35.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -70,10 +64,27 @@ def scalacOptionsFor(scalaBinVersion: String): Seq[String] = scalaBinVersion mat
...
+      case _ => sys.error(s"Cannot find previous versions for ${version.value}")
+    }
+
+    scalaBinaryVersion.value match {
+      case sbv if sbv.equals(scala213) => Set.empty
+      case _ => previousVersions.toSet.map(previousVersion => organization.value %% name.value % previousVersion)
+    }
+  },
+  mimaBinaryIssueFilters ++= Seq(
+    ProblemFilters.exclude[MissingTypesProblem]("play.api.libs.ws.ahc.AhcWSClientConfig$"),
```
Comment (dwijnand) [inline build.sbt]: Where did `AhcWSClientConfig` go?

File: build.sbt [hunk 2]
```diff
@@ -70,10 +64,27 @@ def scalacOptionsFor(scalaBinVersion: String): Seq[String] = scalaBinVersion mat
...
+    ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.libs.ws.ahc.AhcWSClientConfig.apply"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.libs.ws.ahc.AhcWSClientConfig.copy"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.libs.ws.ahc.AhcWSClientConfig.this"),
```
Comment (dwijnand) [inline build.sbt]: Should we avoid these unnecessary binary breakages?

File: build.sbt [hunk 3]
```diff
@@ -70,10 +64,27 @@ def scalacOptionsFor(scalaBinVersion: String): Seq[String] = scalaBinVersion mat
...
+  mimaPreviousArtifacts := {
+    val VersionPattern = """^(\d+).(\d+).(\d+)(-.*)?""".r
+    val previousVersions = version.value match {
+      case VersionPattern(epoch, major, minor, _) => (0 until minor.toInt).map(v => s"$epoch.$major.$v")
+      case _ => sys.error(s"Cannot find previous versions for ${version.value}")
+    }
+
+    scalaBinaryVersion.value match {
+      case sbv if sbv.equals(scala213) => Set.empty
+      case _ => previousVersions.toSet.map(previousVersion => organization.value %% name.value % previousVersion)
```
Comment (dwijnand) [inline build.sbt]: This could be a simpler `if` expression 🙂

Comment (ignasi35) [reply build.sbt]: Agree. Let's merge and fix on a separate PR.

**Comment by ignasi35** [comment]
Backported `2.0.x` 3c7aaa8f2a3a4134a5034f3ad06ee293a220fec1

**Comment by dwijnand** [comment]
Does that mean we intend to ship a binary breakage in Play 2.7?

**Pull request merged**